### PR TITLE
Fallback to java home if it was not available on PATH

### PR DESCRIPTION
**Pull request by tgodzik**
Previously if Java was not available on PATH, the launcher would fail and not start the Bloop server. Now we revert to java home if that was the case.

### CODE REVIEW

File: bloopgun/src/main/scala/bloop/bloopgun/Bloopgun.scala [hunk 1]
```diff
@@ -488,13 +490,40 @@ class BloopgunCli(
...
+      } catch {
+        case e: IOException =>
+          val javaExec = sys.env
+            .get("JAVA_HOME")
+            .orElse(sys.props.get("java.home"))
+            .map(Paths.get(_).resolve("bin/java"))
+            .filter(_.toFile().isFile())
+            .map(_.toString())
+          javaExec match {
+            case Some(java) if e.getMessage().contains("No such file or directory") =>
```
Comment (jvican) [inline bloopgun/src/main/scala/bloop/bloopgun/Bloopgun.scala]: Is this message shown in all platforms?

File: bloopgun/src/main/scala/bloop/bloopgun/Bloopgun.scala [hunk 2]
```diff
@@ -415,7 +416,8 @@ class BloopgunCli(
 
     def cmdWithArgs(
         found: LocatedServer,
-        extraJvmOpts: List[String]
+        extraJvmOpts: List[String],
+        javaCommand: List[String] = List("java")
```
Comment (jvican) [inline bloopgun/src/main/scala/bloop/bloopgun/Bloopgun.scala]: Should we call this `javaBinary`?

Comment (tgodzik) [reply bloopgun/src/main/scala/bloop/bloopgun/Bloopgun.scala]: Fixed!

File: bloopgun/src/main/scala/bloop/bloopgun/Bloopgun.scala (hunk 1 above)
Comment (tgodzik) [reply bloopgun/src/main/scala/bloop/bloopgun/Bloopgun.scala]: Actually checked and it is language dependent, changed to: 
```
val javaErrorMessage = "Cannot run program \"java\""
val errorCode = "error=2"
javaExec match {
 case Some(java)
    if e.getMessage().contains(javaErrorMessage) && e.getMessage.contains(errorCode) =>
```

These two parts seem to be stable in Windows and on Linux. The message itself comes from ProcessBuilder and error code is written by the system:

Windows:
```
Exception in thread "main" java.io.IOException: Cannot run program "javaa": CreateProcess error=2, Nie mo�na odnale�� okre�lonego pliku
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at gradle.test.project.Main$.delayedEndpoint$gradle$test$project$Main$1(Library.scala:20)
	at gradle.test.project.Main$delayedInit$body.apply(Library.scala:13)
	at scala.Function0.apply$mcV$sp(Function0.scala:39)
	at scala.Function0.apply$mcV$sp$(Function0.scala:39)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
	at scala.App.$anonfun$main$1$adapted(App.scala:80)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.App.main(App.scala:80)
	at scala.App.main$(App.scala:78)
	at gradle.test.project.Main$.main(Library.scala:13)
	at gradle.test.project.Main.main(Library.scala)
Caused by: java.io.IOException: CreateProcess error=2, Nie mo�na odnale�� okre�lonego pliku
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 11 more
```

Linux:
```
Exception in thread "main" java.io.IOException: Cannot run program "javaa": error=2, No such file or directory
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at Main$.delayedEndpoint$Main$1(Main.scala:8)
	at Main$delayedInit$body.apply(Main.scala:1)
	at scala.Function0.apply$mcV$sp(Function0.scala:39)
	at scala.Function0.apply$mcV$sp$(Function0.scala:39)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
	at scala.App.$anonfun$main$1$adapted(App.scala:80)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.App.main(App.scala:80)
	at scala.App.main$(App.scala:78)
	at Main$.main(Main.scala:1)
	at Main.main(Main.scala)
Caused by: java.io.IOException: error=2, No such file or directory
	at java.lang.UNIXProcess.forkAndExec(Native Method)
	at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
	at java.lang.ProcessImpl.start(ProcessImpl.java:134)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 11 more
```
Checking on MacOS

Comment (tgodzik) [reply bloopgun/src/main/scala/bloop/bloopgun/Bloopgun.scala]: MacOS looks similar to linux:
```
java.io.IOException: Cannot run program "java": error=2, No such file or directory
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at Main$.delayedEndpoint$Main$1(TomekMain.scala:7)
	at Main$delayedInit$body.apply(TomekMain.scala:1)
	at scala.Function0.apply$mcV$sp(Function0.scala:39)
	at scala.Function0.apply$mcV$sp$(Function0.scala:39)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
	at scala.App.$anonfun$main$1(App.scala:73)
	at scala.App.$anonfun$main$1$adapted(App.scala:73)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at scala.App.main(App.scala:73)
	at scala.App.main$(App.scala:71)
	at Main$.main(TomekMain.scala:1)
	at Main.main(TomekMain.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at scala.reflect.internal.util.ScalaClassLoader.$anonfun$run$2(ScalaClassLoader.scala:105)
	at scala.reflect.internal.util.ScalaClassLoader.asContext(ScalaClassLoader.scala:40)
	at scala.reflect.internal.util.ScalaClassLoader.asContext$(ScalaClassLoader.scala:37)
	at scala.reflect.internal.util.ScalaClassLoader$URLClassLoader.asContext(ScalaClassLoader.scala:130)
	at scala.reflect.internal.util.ScalaClassLoader.run(ScalaClassLoader.scala:105)
	at scala.reflect.internal.util.ScalaClassLoader.run$(ScalaClassLoader.scala:97)
	at scala.reflect.internal.util.ScalaClassLoader$URLClassLoader.run(ScalaClassLoader.scala:130)
	at scala.tools.nsc.CommonRunner.run(ObjectRunner.scala:29)
	at scala.tools.nsc.CommonRunner.run$(ObjectRunner.scala:28)
	at scala.tools.nsc.ObjectRunner$.run(ObjectRunner.scala:43)
	at scala.tools.nsc.CommonRunner.runAndCatch(ObjectRunner.scala:35)
	at scala.tools.nsc.CommonRunner.runAndCatch$(ObjectRunner.scala:34)
	at scala.tools.nsc.MainGenericRunner.runTarget$1(MainGenericRunner.scala:70)
	at scala.tools.nsc.MainGenericRunner.run$1(MainGenericRunner.scala:91)
	at scala.tools.nsc.MainGenericRunner.process(MainGenericRunner.scala:103)
	at scala.tools.nsc.MainGenericRunner$.main(MainGenericRunner.scala:108)
	at scala.tools.nsc.MainGenericRunner.main(MainGenericRunner.scala)
Caused by: java.io.IOException: error=2, No such file or directory
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 36 more
```

Comment (jvican) [reply bloopgun/src/main/scala/bloop/bloopgun/Bloopgun.scala]: Great detective work!

**Pull request merged**